### PR TITLE
[Merged by Bors] - chore: use leanOptions in lakefile

### PR DESCRIPTION
**Pull request by kim-em**
This is a PR to `bump/v4.4.0`, updating to nightly-2023-11-28, and updating the lakefile for the new way of specifying options.

---

The matching PR for Std is at https://github.com/leanprover/std4/pull/406. They can be merged independently.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by tydeu** [comment]
@semorrison Now that combinations are no longer necessary and the options are set globally on the package, why not just inline the definition to to the package configuration? That is, just have:

```lean
package mathlib where
  leanOptions := #[
    ⟨`pp.unicode.fun, true⟩,
    ⟨`pp.proofs.withType, false⟩,
    ⟨`autoImplicit, false⟩,
    ⟨`relaxedAutoImplicit, false⟩
  ]
  weakLeanArgs := weakLeanArgs
```

File: lakefile.lean [hunk 1]
```diff
@@ -2,30 +2,22 @@ import Lake
 
 open Lake DSL
 
-def moreServerArgs := #[
-  "-Dpp.unicode.fun=true", -- pretty-prints `fun a ↦ b`
```
Comment (eric-wieser) [inline lakefile.lean:6]: Should we keep this comment, or is it boring? Does hovering over `` `pp.unicode.fun`` display the docstring for the option?

Comment (semorrison) [reply lakefile.lean:6]: Can we worry about that later, or separately?

Comment (semorrison) [reply lakefile.lean:6]: Hovering says "A literal of type `Name`."

Comment (eric-wieser) [reply lakefile.lean:6]: Let's keep the comment for now then; clearly something thought it was useful, and we can decide to remove it later separately.

Comment (eric-wieser) [reply lakefile.lean:6]: (but pinging @tydeu about the feature request of option docstrings in vscode hovers here)

File: lakefile.lean [hunk 2]
```diff
@@ -2,30 +2,22 @@ import Lake
...
-    #["-DwarningAsError=true"]
-  else
-    #[]
-
 package mathlib where
-  moreServerArgs := moreServerArgs
-  moreLeanArgs := moreLeanArgs
-  weakLeanArgs := weakLeanArgs
+  leanOptions := #[
+    ⟨`pp.unicode.fun, true⟩,
```
Comment (eric-wieser) [inline lakefile.lean:7]:
```suggestion
    ⟨`pp.unicode.fun, true⟩, -- pretty-prints `fun a ↦ b`
```

**Review by eric-wieser — APPROVED** [review]
bors d+

This is certainly much cleaner, but perhaps it makes sense to get an approval from @tydeu who can comment if this is correct.

**Review by tydeu — APPROVED** [review]
LGTM!

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
I think #8758 will need to go in first.

**Comment by semorrison** [comment]
bors p=10

bors merge

**Pull request closed without merging**